analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp7.cs

```diff
@@ -334,4 +334,48 @@ public int Foo()
             return 1;
         }
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/3910
+    public class Repro_3910
+    {
+        public void ElementAccess1(Exception ex)
+        {
+            if (ex?.Data["Key"] is string value)
+            {
+            }
+            else if (ex != null) // Noncompliant FP
+            {
+            }
+        }
+
+        public void ElementAccess2(Exception ex)
+        {
+            if (ex?.Data["Key"] is string value)
+            {
+            }
+            else if (ex.Data != null) // FN
```

Comment:
in this case, this is relying on the fact that if ex.Data is null, it will throw an exception at line 353, right? 

could you please add a comment with this, because I have to use pen and paper to figure it out?

Reply:
Good point, this is actually a bad test case.